### PR TITLE
Delete server side liveshare handlers.

### PR DESCRIPTION
**Pull request by dibarbet**
Delete the serverside handlers for liveshare requests that were migrated to ILanguageClient.

Typescript handlers were slightly refactored + merged into one location.

### CODE REVIEW

File: src/Features/LanguageServer/Protocol/Handler/Formatting/FormatDocumentHandlerBase.cs [hunk 1]
```diff
@@ -13,28 +14,33 @@
 
 namespace Microsoft.CodeAnalysis.LanguageServer.Handler
 {
-    internal class FormatDocumentHandlerBase
+    internal abstract class FormatDocumentHandlerBase
```
Comment (jasonmalinowski) [inline src/Features/LanguageServer/Protocol/Handler/Formatting/FormatDocumentHandlerBase.cs]: We're confident nobody was creating this directly?

File: src/VisualStudio/LiveShare/Impl/LoadHandler.cs [hunk 2]
```diff
@@ -1,39 +0,0 @@
...
-
-using System;
-using System.Threading;
-using System.Threading.Tasks;
-using Microsoft.VisualStudio.LiveShare.LanguageServices;
-using CustomMethods = Microsoft.VisualStudio.LiveShare.LanguageServices.Protocol.CustomMethods;
-
-namespace Microsoft.VisualStudio.LanguageServices.LiveShare
-{
-    internal class LoadHandler : ILspRequestHandler, ILspRequestHandler<object, object>
```
Comment (jasonmalinowski) [inline src/VisualStudio/LiveShare/Impl/LoadHandler.cs]: What was this originally used for? If old clients request it, will they gracefully handle it missing? Or do we just not care anymore? (I'm OK with not caring.)

File: src/VisualStudio/LiveShare/Impl/Shims/TypeScriptHandlerShims.cs [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+        }
+
+        public Task<TextEdit[]> HandleAsync(DocumentRangeFormattingParams request, RequestContext<Solution> requestContext, CancellationToken cancellationToken)
+            => base.HandleRequestAsync(requestContext.Context, request, requestContext.ClientCapabilities?.ToObject<ClientCapabilities>(), cancellationToken);
+
+        protected override async Task<IList<TextChange>> GetFormattingChangesAsync(IEditorFormattingService formattingService, Document document, TextSpan? textSpan, CancellationToken cancellationToken)
+        {
+            // TypeScript expects to be called on the UI thread to get the formatting options.
+            await _threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
+            return await base.GetFormattingChangesAsync(formattingService, document, textSpan, cancellationToken).ConfigureAwait(false);
```
Comment (jasonmalinowski) [inline src/VisualStudio/LiveShare/Impl/Shims/TypeScriptHandlerShims.cs]: If this is following JTF rules we don't really need the ConfigureAwait(false) here.

File: src/VisualStudio/LiveShare/Impl/Shims/TypeScriptHandlerShims.cs [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+            {
+                // If the value is true, set it to false.  Liveshare does not support hierarchical document symbols.
+                clientCapabilities.TextDocument.DocumentSymbol.HierarchicalDocumentSymbolSupport = false;
+            }
+
+            var handler = (IRequestHandler<DocumentSymbolParams, object[]>)LazyRequestHandler.Value;
+            var response = await handler.HandleRequestAsync(requestContext.Context, param, clientCapabilities, cancellationToken).ConfigureAwait(false);
+
+            // Since hierarchicalSupport will never be true, it is safe to cast the response to SymbolInformation[]
+            return response.Select(obj => (SymbolInformation)obj).ToArray();
```
Comment (jasonmalinowski) [inline src/VisualStudio/LiveShare/Impl/Shims/TypeScriptHandlerShims.cs]: There's `.Cast<SymbolInformation>()` you can use.

File: src/VisualStudio/LiveShare/Impl/LoadHandler.cs (hunk 2 above)
Comment (dibarbet) [reply src/VisualStudio/LiveShare/Impl/LoadHandler.cs]: It used to be used to trigger loads of language servers in liveshare.  However talked to Dayton and they confirmed it's no longer used.

File: src/Features/LanguageServer/Protocol/Handler/Formatting/FormatDocumentHandlerBase.cs (hunk 1 above)
Comment (dibarbet) [reply src/Features/LanguageServer/Protocol/Handler/Formatting/FormatDocumentHandlerBase.cs]: yes, no one outside of Roslyn has IVTs to this project

File: src/VisualStudio/LiveShare/Impl/Shims/TypeScriptHandlerShims.cs (hunk 3 above)
Comment (dibarbet) [reply src/VisualStudio/LiveShare/Impl/Shims/TypeScriptHandlerShims.cs]: There's a warning if I leave it out.  I don't think it should be true though right?

Comment (jasonmalinowski) [reply src/VisualStudio/LiveShare/Impl/Shims/TypeScriptHandlerShims.cs]: Yeah, probably not worth fighting. We could disable the warning but this is all going away.

**Pull request merged**